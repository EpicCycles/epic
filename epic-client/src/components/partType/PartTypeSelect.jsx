import React from "react";

import SelectInput from "../../common/SelectInput";

const PartTypeSelect = (props) => {
    const { sections, fieldName, onChange, partTypeSelected, isEmptyAllowed, error } = props;
    const partTypeOptions = [];
    sections && sections.forEach(section => {
        partTypeOptions.concat(section.partTypes.map(partType => {
            return {
                'value': partType.id ? partType.id : partType.dummyKey,
                'name': partType.shortName + " (" + section.shortName + ")"
            };
        }));
    });
    return <SelectInput
        fieldName={fieldName}
        onChange={onChange}
        value={partTypeSelected}
        options={partTypeOptions}
        isEmptyAllowed={isEmptyAllowed}
        error={error}
    />;
};
export default PartTypeSelect;