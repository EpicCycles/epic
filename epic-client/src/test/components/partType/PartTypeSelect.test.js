import React, {Fragment} from "react";
import PartTypeSelect from "../../../components/partType/PartTypeSelect";

test("it displays correctly", () => {
    const sections = [
        {
            id: 1,
            shortName: "sections1",
            partTypes: [
                {
                    id: 11, shortName: "partType 1",
                }
            ]
        },
        {
            id: 2,
            shortName: "sections2",
            partTypes: [
                {
                    id: 21, shortName: "partType 21",
                },
                {
                    id: 22, shortName: "partType 22",
                },
                {
                    id: 23, shortName: "partType 23",
                },
            ]
        },
    ];
    const component = shallow(<PartTypeSelect
        sections={sections}
        fieldName="name_of_field"
    />);
    expect(component).toMatchSnapshot();
});