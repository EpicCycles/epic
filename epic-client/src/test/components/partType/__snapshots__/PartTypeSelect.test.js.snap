// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it displays correctly 1`] = `
<SelectInput
  fieldName="name_of_field"
  options={Array []}
/>
`;
